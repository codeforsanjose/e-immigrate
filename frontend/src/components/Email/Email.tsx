import React from 'react';
import './Email.css';
import { CommonComponentProps } from '../../types/CommonComponentProps';
import { useContentContext } from '../../contexts/ContentContext';
import { RequiredErrorDiv } from '../RequiredErrorPresenter';
import { useQuestionnaireResponseContent } from '../../contexts/QuestionnaireResponseContext';

type EmailProps = CommonComponentProps;
export function Email(props: EmailProps) {
    const {
        q, 
    } = props;
    const { content } = useContentContext();
    const { 
        collectAnswer,
        bindField,
    } = useQuestionnaireResponseContent();
    return (
        <>
            <input
                type="email"
                id={q.slug}
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <RequiredErrorDiv message={content.errorMessageEmail} />
        </>
    );
}
