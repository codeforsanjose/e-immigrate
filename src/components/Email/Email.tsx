import React from 'react';
import './Email.css';
import { BindFieldFunction, CollectAnswerFunction, ContentData, QData } from '../../types/common';

type EmailProps = {
    attributes: {
        q: QData;
        bindField: BindFieldFunction; 
        content: {
            errorMessageEmail: string;
        };
        collectAnswer: CollectAnswerFunction;
    };
};
export function Email(props: EmailProps) {
    const {
        attributes: { 
            q, 
            bindField, 
            collectAnswer, 
            content,
        },
    } = props;
    return (
        <>
            <input
                type="email"
                id={q.slug}
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <div className="RequiredError">*{content.errorMessageEmail}</div>
        </>
    );
}
