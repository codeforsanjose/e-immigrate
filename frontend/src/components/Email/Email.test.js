import React from 'react';
import Email from './Email';

import renderer from 'react-test-renderer';

it('Email renders correctly', () => {
    const tree = renderer
        .create(
            <Email
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
