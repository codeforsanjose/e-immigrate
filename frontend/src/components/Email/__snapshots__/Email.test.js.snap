// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email renders correctly 1`] = `
Array [
  <input
    className="TextInput"
    onChange={[Function]}
    type="email"
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
